File: core/src/main/java/com/example/anders/geoquiz/models/QuestionGenerator.java
package com.example.anders.geoquiz.models;

/**
 * Created by anders on 18/12/14.
 */
public interface QuestionGenerator {
    TrueFalseQuestion generateStartingQuestion();

    TrueFalseQuestion generateNextQuestion();

    TrueFalseQuestion generatePreviousQuestion();

    int getCurrentQuestionIndex();
}

File: core/src/main/java/com/andersmurphy/geoquiz/models/QuestionGenerator.kt
package com.andersmurphy.geoquiz.models

/**
 * Created by anders on 18/12/14.
 */
interface QuestionGenerator {
    fun generateStartingQuestion(): TrueFalseQuestion
    fun generateNextQuestion(): TrueFalseQuestion
    fun generatePreviousQuestion(): TrueFalseQuestion
    fun getCurrentQuestionIndex(): Int
}
